src/components/Chat/InputMessageV2.tsx

```diff
@@ -0,0 +1,219 @@
+  emojiColor,
+  profileImage,
+  username,
+  onToggleConfirm,
+  type,
+  isShowToggle = true,
+  messageDisable
+}: InputMessageV2Props) => {
+  const refEmoji = React.useRef(null);
+  const [inputFocus, setInputFocus] = React.useState(false);
+  const [text, setText] = React.useState('');
+
+  const onChangeInput = (v) => {
+    setText(v);
+  };
+  const onSelectEmoji = () => {
+    refEmoji.current.close();
+  };
+
+  const handleSendMessage = () => {
+    onSendButtonClicked(text);
+    setText('');
+  };
+
+  const isDisableButton = () => {
+    return text?.length === 0;
+  };
+
+  const sendButtonStyle = React.useCallback(() => {
+    const isDisabled = isDisableButton();
+    if (isDisabled) return colors.gray1;
+    if (type === 'SIGNED') return colors.darkBlue;
+    return colors.bondi_blue;
+  }, [isDisableButton()]);
+
+  const toggleChnage = () => {
+    if (messageDisable) {
+      ToastMessage.show({
+        type: 'asNative',
+        text1: messageDisable,
+        position: 'bottom'
+      });
+      return;
+    }
+
+    Alert.alert(
+      '',
+      type === 'ANONYMOUS'
+        ? `Switch back to regular chat?\nMessage ${username} using your username.`
+        : 'Switch to anonymous chat?\nMessage this user anonymously instead.',
+      [
+        {
+          text: 'Cancel'
+        },
+        {text: 'Yes, move to other chat', onPress: onToggleConfirm}
+      ]
+    );
+  };
+
+  return (
+    <View style={styles.main}>
+      {emojiCode ? (
+        <View style={[styles.btnEmoji, {backgroundColor: emojiColor}]}>
+          <Text style={{textAlign: 'center', textAlignVertical: 'center'}}>{emojiCode}</Text>
+        </View>
+      ) : (
+        <Image source={{uri: profileImage}} style={styles.btnEmoji} />
+      )}
+      <View style={styles.containerInput}>
+        <TextInput
+          multiline
+          style={styles.input}
+          onChangeText={onChangeInput}
+          value={text}
+          onFocus={() => setInputFocus(true)}
+          onBlur={() => setInputFocus(false)}
+          numberOfLines={4}
+        />
+        {isShowToggle && (
+          <ToggleSwitch
+            labelLeft={!inputFocus ? 'Anonymity' : undefined}
+            styleLabelLeft={[
+              styles.labelToggle,
+              {color: type === 'ANONYMOUS' ? colors.gray : colors.blue}
+            ]}
+            onValueChange={toggleChnage}
+            value={type === 'ANONYMOUS'}
+          />
+        )}
+      </View>
+      <TouchableOpacity
+        style={[styles.btn, isDisableButton() ? styles.disableButton : styles.enableButton]}
+        disabled={isDisableButton()}
+        onPress={handleSendMessage}>
+        <IconSend
+          style={styles.icSendButton}
+          fillBackground={sendButtonStyle()}
+          fillIcon={colors.white}
+        />
+      </TouchableOpacity>
+      <SheetEmoji ref={refEmoji} selectEmoji={onSelectEmoji} />
+    </View>
+  );
+};
+
+export default React.memo(InputMessageV2);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `InputMessageV2` component seems to be well implemented. It includes proper type definitions, state management, and event handlers. The styles are defined in a clear and organized manner. However, there is one potential issue related to the `sendButtonStyle` function.



<!-- This is an auto-generated comment by OSS CodeRabbit -->